File: Web/Payroll.Shared/Data/FnFSettlement.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Payroll.Shared.Data
{
    [Table("fnf_settlements")]
    public class FnFSettlement
    {
        [Key]
        [Column("settlement_id")]
        public int SettlementId { get; set; }

        [Required]
        [Column("employee_id")]
        public int EmployeeId { get; set; }

        [Required]
        [Column("resignation_request_id")]
        public int ResignationRequestId { get; set; }

        [Column("settlement_date")]
        public DateTime SettlementDate { get; set; } = DateTime.Now;

        // --- EARNINGS ---
        [Column("unpaid_salary")]
        public decimal UnpaidSalary { get; set; } // Salary for days worked in last month

        [Column("leave_encashment_amount")]
        public decimal LeaveEncashment { get; set; }

        [Column("gratuity_amount")]
        public decimal Gratuity { get; set; }

        [Column("bonus_payable")]
        public decimal BonusPayable { get; set; }

        // --- DEDUCTIONS ---
        [Column("notice_period_recovery")]
        public decimal NoticePeriodRecovery { get; set; } // If they leave early

        [Column("asset_recovery_cost")]
        public decimal AssetRecoveryCost { get; set; } // Damaged laptop, etc.

        [Column("outstanding_advances")]
        public decimal OutstandingAdvances { get; set; }

        // --- NET ---
        [Column("net_payable")]
        public decimal NetPayable { get; set; }

        [Column("is_finalized")]
        public bool IsFinalized { get; set; } = false;

        [Column("payment_reference")]
        public string? PaymentReference { get; set; } // Check No / UTR
    }
}

File: Android/app/src/main/java/com/biometric/app/api/FnFSettlement.kt
package com.biometric.app.api

import androidx.annotation.Keep
import com.google.gson.annotations.SerializedName

@Keep
data class FnFSettlement(
    @SerializedName("settlementId") val settlementId: Int = 0,
    @SerializedName("employeeId") val employeeId: Int = 0,
    @SerializedName("resignationRequestId") val resignationRequestId: Int = 0,
    @SerializedName("settlementDate") val settlementDate: String? = null,
    
    // --- EARNINGS ---
    @SerializedName("unpaidSalary") var unpaidSalary: Double = 0.0,
    @SerializedName("leaveEncashment") var leaveEncashment: Double = 0.0,
    @SerializedName("gratuity") var gratuity: Double = 0.0,
    @SerializedName("bonusPayable") var bonusPayable: Double = 0.0,
    
    // --- DEDUCTIONS ---
    @SerializedName("noticePeriodRecovery") var noticePeriodRecovery: Double = 0.0,
    @SerializedName("assetRecoveryCost") var assetRecoveryCost: Double = 0.0,
    @SerializedName("outstandingAdvances") var outstandingAdvances: Double = 0.0,
    
    // --- NET ---
    @SerializedName("netPayable") var netPayable: Double = 0.0,
    @SerializedName("isFinalized") var isFinalized: Boolean = false,
    @SerializedName("paymentReference") var paymentReference: String? = null
)
